fuel-core/src/tx_pool.rs

```diff
@@ -47,20 +50,32 @@ impl From<crate::state::Error> for Error {
     }
 }
 
+impl TxPoolDB for Database {}
+
 /// Holds submitted transactions and attempts to propose blocks
 pub struct TxPool {
     executor: Executor,
     db: Database,
+    fuel_txpool: Arc<dyn TxPoolTrait>,
```

Comment:
Does this need to be internally arc'd if the `TxPool` struct is already arc'd? Maybe we could get away with a box instead?

Reply:
Probably not needed, will see to change it.